src/amp-story-embed.js

```diff
@@ -31,6 +33,58 @@ const CSS = `
   main { display: flex; flex-direction: row; height: 100%; }
   .i-amphtml-story-embed-loaded iframe { opacity: 1; }`;
 
+/** @const {string} */
+const TAG = '[AMP-STORY-EMBED]';
+
+/**
+ * Logging.
+ */
+function log() {
+  const var_args = toArray(arguments);
+  var_args.unshift(TAG);
+  console /*OK*/.log
+    .apply(console, var_args);
+}
+
+/**
+ * Gets encoded url for viewer usage.
+ * @param {string} href
+ * @return {!Location}
+ */
+function getEncodedUrl(href) {
+  const params = {
+    visibilityState: 'inactive',
+    origin: parseUrlDeprecated(self.location.href).origin,
+  };
+  log('Params: ' + JSON.stringify(params));
+
+  let inputUrl = href + '?amp_js_v=0.1#' + encodeUrl(params);
+  if (self.location.hash && self.location.hash.length > 1) {
+    inputUrl += '&' + self.location.hash.substring(1);
+  }
+  return parseUrlDeprecated(inputUrl);
+}
+
+/**
+ * Encodes string of URL parameters.
+ * @param {!Object} params
+ * @return {string}
+ */
+function encodeUrl(params) {
```

Comment:
Can't you use this util instead of writing your own?

https://github.com/ampproject/amphtml/blob/2d23dd4bedd6c0bd1af8664ca82c8120a2595247/src/url.js#L262-L279